examples/rtic_spi.rs

```diff
@@ -4,64 +4,108 @@
-        });
-        // Starts the I/O as soon as we're done initializing, since
-        // the TX FIFO is empty.
-        spi.set_interrupts(Interrupts::TRANSMIT_DATA);
-        (Shared {}, Local { spi }, init::Monotonics())
-    }
+    #[init(local = [
+        spi_systick: Option<Systick> = None,
+    ])]
+    fn init(cx: init::Context) -> (Shared, Local) {
+        let (
+            board::Common { mut dma, .. },
+            board::Specifics {
+                spi: (mut spi_bus, spi_cs_pin),
+                ..
+            },
+        ) = board::new();
 
-    #[task(binds = BOARD_SPI, local = [spi])]
-    fn spi_interrupt(cx: spi_interrupt::Context) {
-        let spi_interrupt::LocalResources { spi } = cx.local;
+        // Init monotonic
+        let systick_token = rtic_monotonics::create_systick_token!();
+        Systick::start(
+            cx.core.SYST,
+            600_000_000, /* TODO: fix */
+            systick_token,
+        );
+
+        // Init DMA
+        let mut chan_a = dma[board::BOARD_DMA_A_INDEX].take().unwrap();
+        chan_a.set_disable_on_completion(true);
 
-        let status = spi.status();
-        spi.clear_status(Status::TRANSMIT_DATA | Status::RECEIVE_DATA);
+        let mut chan_b = dma[board::BOARD_DMA_B_INDEX].take().unwrap();
+        chan_b.set_disable_on_completion(true);
 
-        if status.intersects(Status::TRANSMIT_DATA) {
-            // This write clears TRANSMIT_DATA.
-            spi.set_interrupts(Interrupts::RECEIVE_DATA);
+        // Configure SPI
+        spi_bus.set_dma(LpspiDma::Full(chan_a, chan_b));
+        let spi_interrupt_handler = spi_bus.enable_interrupts();
 
-            // Sending two u32s. Frame size is represented by bits.
-            let transaction = Transaction::new(2 * 8 * core::mem::size_of::<u32>() as u16)
-                .expect("Transaction frame size is within bounds");
-            spi.enqueue_transaction(&transaction);
+        // Create SPI device
+        let spi_device = ExclusiveDevice::new(spi_bus, spi_cs_pin, Systick);
 
-            spi.enqueue_data(0xDEADBEEF);
-            spi.enqueue_data(!0xDEADBEEF);
-        } else if status.intersects(Status::RECEIVE_DATA) {
-            // This write clears RECEIVE_DATA.
-            spi.set_interrupts(Interrupts::TRANSMIT_DATA);
+        demo::spawn().unwrap();
 
-            assert!(spi.fifo_status().rxcount == 2);
+        (
+            Shared {},
+            Local {
+                spi_device,
+                spi_interrupt_handler,
+            },
+        )
+    }
 
-            while let Some(_) = spi.read_data() {}
+    #[task(priority = 1, local = [spi_device])]
+    async fn demo(cx: demo::Context) {
+        let demo::LocalResources { spi_device, .. } = cx.local;
+
+        loop {
+            Systick::delay(1000.millis()).await;
+
+            // To demonstrate normal operation
+            spi_device
+                .transaction(&mut [
+                    Operation::DelayUs(100),
```

Comment:
Won't these sorts of delays effectively imply a timer queue and async executor exists?